### ISSUE TITLE
[REVIEW]: NetworkSedimentTransporter: A Landlab submodule for bed material transport through river networks

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2341]
**Submitting author:** @pfeiffea (<a href="http://orcid.org/0000-0002-3974-132X">Allison Pfeiffer</a>)
**Repository:** <a href="https://github.com/landlab/landlab" target ="_blank">https://github.com/landlab/landlab</a>
**Version:** v2.2.0
**Editor:** @kthyng
**Reviewer:** @zsylvester, @ebgoldstein
**Archive:** <a href="https://doi.org/10.5281/zenodo.4044683" target="_blank">10.5281/zenodo.4044683</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8/status.svg)](https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8"><img src="https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8/status.svg)](https://joss.theoj.org/papers/b76b97c3ddacfafdea0892ae82bbc6e8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/landlab/landlab" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@zsylvester & @ebgoldstein, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kthyng know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/landlab/landlab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@pfeiffea) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/landlab/landlab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@pfeiffea) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @zsylvester, @ebgoldstein it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02341/joss.02341/10.21105.joss.02341.pdf)

**Comment by kthyng** [comment]
Hi @zsylvester, @ebgoldstein, just want to send a friendly ping to keep this review on your radar for the next few weeks. Thanks!

**Comment by ebgoldstein** [comment]
Hi @pfeiffea — I'm walking through the tutorials now. They are very good. Two things so far: 

1) in the index to Landlab notebooks, Network Sediment Transporter is listed twice. Here is a screenshot: 
<img width="432" alt="Screen Shot 2020-07-13 at 11 39 56 AM" src="https://user-images.githubusercontent.com/5330599/87324370-ff774d00-c4fd-11ea-96b8-c73a230d3c0e.png">

2) I have walking through the NST tutorials using a Binder instance . Some of the links (specifically to other tutorials) point to a local port —  `localhost:8888` — and therefore don't function in my Binder instance.

**Comment by ebgoldstein** [comment]
Ok, I have made it through the paper, docs, tutorials, and repo. The code works as described, and is well served by 4 tutorial notebooks. I think users will be able to understand the working of the code and how to modify it for their given use case. To my eye, this is a nice addition to Landlab,

I have one suggestion: The module described is in some sense a computational replication of Czuba 2018 — and i noticed that the tutorial for a real basin was a section of the the Methow River, which is the same as the original paper. I know this is JOSS and not Rescience C, but I think a mention of the results of this replication could be made. For example, the the authors could show (as a figure) or describe (in text) or put in the tutorials a statement about the comparison to Czuba 2018 Fig2 or Fig3... even a simple statement that 'the results are identical to the code of C18'.   I will leave this to @kthyng to decide if it is in scope. 

also: 
> Nothing will stop us from making all of the choices at once. 
😆

**Comment by kthyng** [comment]
@pfeiffea Just want to note you have some reviewer comments above. 

> I have one suggestion: The module described is in some sense a computational replication of Czuba 2018 — and i noticed that the tutorial for a real basin was a section of the the Methow River, which is the same as the original paper. I know this is JOSS and not Rescience C, but I think a mention of the results of this replication could be made. For example, the the authors could show (as a figure) or describe (in text) or put in the tutorials a statement about the comparison to Czuba 2018 Fig2 or Fig3... even a simple statement that 'the results are identical to the code of C18'. I will leave this to @kthyng to decide if it is in scope.

@ebgoldstein and @pfeiffea — This sounds like a small and reasonable request, but happy to hear if otherwise.

**Comment by kthyng** [comment]
@zsylvester Would you be able to work on this review in the next two weeks?

**Comment by zsylvester** [comment]
Yes, I will try to get to it this week. I just realized that I have not 'officially' accepted the invitation yet - when I click on the link (https://github.com/openjournals/joss-reviews/invitations) it says that it has expired.

**Comment by zsylvester** [comment]
@kthyng It still does not let me to accept the invitation / go through the checklist.

**Comment by kthyng** [comment]
@whedon re-invite @zsylvester as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@zsylvester please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by kthyng** [comment]
@zsylvester This should do it for you, try the link above.

**Comment by zsylvester** [comment]
I have gone through the paper, the four Jupyter notebooks, scanned the repo, looked at the documentation, and ran the tests.  I think this is a nice addition to the Landlab package and it is likely to be of interest to anybody who is working on sediment transport problems in primarily non-depositional landscapes. The NetworkSedimentTransporter component is installed along with the Landlab package and I had no issues installing Landlab 2.0 (that happened a while ago). The three notebooks ran smoothly on Binder and they are easy to follow along, I can see how the second one could form a starting point for playing with someone's own dataset. I have run the fourth one locally.

Here are a few comments / suggestions:
- I agree with @ebgoldstein that a statement about how the model compares with Czuba (2018) would be useful.
- Along the same lines, it would be nice to see a bit larger / more complex model in the notebook with the shapefile example, maybe something similar to what is shown in Czuba (2018), and show how it could be used to provide insights into some aspects of sediment transport.
- The profiling and scaling notebook could use a bit more explanation / documentation.
- Many of the links in the notebooks do not work.
- Link to Jupyter notebook at the end of the manuscript does not work.
- I think 'MATLAB' is the correct usage, not 'Matlab'.
- 'computational infrastructure built in order to create the NetworkSedimentTransporter' instead of  'computational infrastructure created in order to create the NetworkSedimentTransporter'.
- '...required the addition of two new data structures' instead of '...required the creation of two new data structures'.
- 'Landlab' instead of 'landlab' in the Hobley et al. reference.

**Comment by kthyng** [comment]
@pfeiffea How are your changes coming along? I think the reviews are waiting on you currently.

**Comment by pfeiffea** [comment]
@kthyng Changes are nearly complete-- we're just waiting for them to be published in the newest release of landlab/landlab. We should have a new draft to you all in the next few days.

**Comment by pfeiffea** [comment]
@kthyng, @ebgoldstein and @zsylvester: 

Thank you for your thorough reading and suggestions on the first draft of our submission. We have edited the component, notebooks, and text. The updated version is now published on the main branch of Landlab (v2.2.0): https://github.com/landlab/landlab 

Our response to individual comments are included below. 

**A)** Computational Replication / Comparison to Czuba (2018)

Both reviewers bring up making a clearer comparison between this paper and the work presented in [Czuba (2018)](https://www.sciencedirect.com/science/article/abs/pii/S0169555X18303349?via%3Dihub). Specifically the following two comments:

From @ebgoldstein

> I have one suggestion: The module described is in some sense a computational replication of Czuba 2018 — and i noticed that the tutorial for a real basin was a section of the the Methow River, which is the same as the original paper. I know this is JOSS and not Rescience C, but I think a mention of the results of this replication could be made. For example, the the authors could show (as a figure) or describe (in text) or put in the tutorials a statement about the comparison to Czuba 2018 Fig2 or Fig3... even a simple statement that 'the results are identical to the code of C18'. I will leave this to @kthyng to decide if it is in scope.

From @zsylvester:

> I agree with @ebgoldstein that a statement about how the model compares with Czuba (2018) would be useful.

**Response**:
The NetworkSedimentTransporter is not a direct translation of the model presented in Czuba (2018) and thus we don't think an attempt at replicating the results of this paper makes sense. However, we interpret these comments to indicate two important points: (a) that our original text does not make clear that this code is not a direct translation of Czuba (2018) and (b) the specific differences between this implementation and that of Czuba (2018) are not sufficiently clear.

We felt that the best way to address this was to add the following text:

> Our implementation is not a direct translation of the model implemented in MATLAB and described in @Czuba2018. In addition, we add two new elements to the model: variable sediment density and bed-material abrasion, and specification of active layer thickness.

We also revised the text describing the differences for clarity. 




**B)** Profiling and Scaling Notebook

@zsylvester pointed out the bare-bones nature of the profiling and scaling notebook: 

> The profiling and scaling notebook could use a bit more explanation / documentation.

Thank you for pointing out this oversight. We have greatly expanded this notebook. It should be far more helpful to users in its revised form. 

**C)** A More Complex Model Notebook

@zsylvester made a request for a notebook demonstrating a more sophisticated use case:

>[It] would be nice to see a bit larger / more complex model in the notebook with the shapefile example, maybe something similar to what is shown in Czuba (2018), and show how it could be used to provide insights into some aspects of sediment transport.

We feel that a more complex model would be out of scope for a Landlab tutorial. The goal of the Landlab tutorials is to demonstrate a simple, generalized use case of each component, rather than to explore specific scientific questions. Previous work on network-scale Lagrangian sediment transport has demonstrated the value of these models to our understanding of basin scale sediment dynamics (see [Gran and Czuba, 2017](http://geomorphology.sese.asu.edu/Papers/Gran_2017_sediment_pulse.pdf) , [Czuba and Foufoula-Georgiou, 2015](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2014WR016139) ). Furthermore, the updated profiling and scaling notebook will provide potential users a sense for the efficiency of this component for larger models. 

**D)** New Active Layer Options

The original version of the paper text referenced multiple methods for determining active layer thickness:

> Many sediment transport models [e.g., @Cui2007b; @Czuba2018] represent the mobile portion of the grains on the riverbed at any given time as an "active layer" of constant thickness. All grains in this layer are transported, whereas all grains below this layer are immobile. Within ``NetworkSedimentTransporter`` the user has the option to specify active layer thickness as a constant value or a multiple of the mean grain size in each link. Alternatively, we incorporated the formulation of @Wongetal2007 to calculate an active layer thickness for each link in the network at each timestep as a function of Shields stress and median grain diameter.

We realized that this functionality was not included in the version of the code that was originally merged into Landlab and submitted to JOSS. We have included this ‘knob’ (and associated tests) in this round of edits. 

**E)** Term used to describe NST

After a discussion between authors, we have decided to call the ``NetworkSedimentTransporter`` a “component” instead of “submodule”. The difference is semantic, though we think that “component” may be clearer for readers. 

**F)** Minor Edits

We have fixed the typos, duplicate notebook listings, and broken links pointed out by @ebgoldstein and @zsylvester. Thank you for finding these!

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02341/joss.02341/10.21105.joss.02341.pdf)

**Comment by kbarnhart** [comment]
@pfeiffea I just checked the PDF and found a few formatting things to fix (e.g., reference capitalization, @mcflugen's  affiliation was incorrect). Am fixing it and will update here when done.

**Comment by kbarnhart** [comment]
@whedon generate pdf from branch barnhark/small_joss_fixes_nst

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch barnhark/small_joss_fixes_nst. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02341/joss.02341/10.21105.joss.02341.pdf)

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02341/joss.02341/10.21105.joss.02341.pdf)

**Comment by kbarnhart** [comment]
OK @pfeiffea I've fixed the small formatting things (was at landlab/landlab/pull/1235)

**Comment by pfeiffea** [comment]
Hi @kthyng -- Let me know if you need anything else from us in terms of a resubmission. I'm still new to this whole github-based review process!

**Comment by kthyng** [comment]
@pfeiffea Looks like you've addressed reviewer comments. I'd like to check in with them.

@zsylvester, @ebgoldstein: Please see comments from @pfeiffea above and let us know if you have more comments or if you are satisfied in the responses. Thanks!

**Comment by ebgoldstein** [comment]
the responses, paper changes, and tutorial notebook changes all look good to me. I have no more comments.

**Comment by zsylvester** [comment]
It all looks great! Now I understand better the differences between this model and the one in Czuba (2018). Only a few minor comments:

- The 'Repository' link on the first page links to the Landlab repo; wouldn't it be better if it linked to [NST](https://github.com/landlab/landlab/tree/master/landlab/components/network_sediment_transporter)?
- There is no URL associated with 'Archive' (page 1)
- 'open-source Python package' instead of 'Open Source Python package' (page 2)
- capitalize 'python' in 'Landlab python package' (page 3)

**Comment by kbarnhart** [comment]
I've addressed the second two comments identified by @zsylvester on landlab/landlab/pull/1237. Should get merged later today after CI runs. 

My answers to the first two are as one of the authors of this paper but reflect knowledge that I have of the JOSS pdf-building and review process that come from being a topical editor.

* I'm not sure whether its possible to point the repository link to 
  https://github.com/landlab/landlab/tree/master/landlab/components/network_sediment_transporter
  instead of to 
  https://github.com/landlab/landlab
  (probably a question for @arfon?)

* I don't think the Archive link should work yet because we haven't yet made an archive (waiting until the end) and no archive has been set.

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02341/joss.02341/10.21105.joss.02341.pdf)

**Comment by kbarnhart** [comment]
Updates addressing half of @zsylvester 's comments now merged into master.

**Comment by pfeiffea** [comment]
@kthyng let us know if these changes adequately address @zsylvester 's comments. 

If so, what are our next steps?

**Comment by kthyng** [comment]
@pfeiffea Sorry I missed this before.

But, it's also not up to me to decide if you have addressed reviewer comments, it is up to the reviewer. @zsylvester how are things looking from your perspective?

**Comment by zsylvester** [comment]
@kthyng @pfeiffea Everything looks fine to me!

**Comment by kthyng** [comment]
Ok then, it looks like the reviewers have both recommended this publication for acceptance. We can start the final bits to wrap up now.

**Comment by kthyng** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5194/esurf-5-21-2017 is OK
- 10.5194/esurf-2020-12 is OK
- 10.1016/j.geomorph.2018.08.031 is OK
- 10.1002/esp.4195 is OK
- 10.1002/2014WR016862 is OK
- 10.1002/2016JF003965 is OK
- 10.1002/esp.4635 is OK
- 10.1080/00221686.2006.9521683 is OK
- 10.1080/00221686.2006.9521684 is OK
- 10.1002/rra.1012 is OK
- 10.1061/(ASCE)0733-9429(2003)129:2(120) is OK
- 10.1029/2006WR005330 is OK
- 10.1029/97WR02387 is OK
- 10.1002/2015JF003491 is OK
- 10.1029/2006WR005172 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02341/joss.02341/10.21105.joss.02341.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02341/joss.02341/10.21105.joss.02341.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
Paper looks great!

Next, @pfeiffea can you verify the version of software you want tagged with this publication?

After that, please upload an archive of your software to something like Zenodo. When doing so, please make sure that the title of author list of the archive exactly match your JOSS paper. Then, report back here with the doi.

**Comment by pfeiffea** [comment]
Thanks, @kthyng!

Landlab v2.2.0

Here's the doi for a Zenodo archive: 10.5281/zenodo.4044683

**Comment by kthyng** [comment]
@whedon set v2.2.0 as version

**Comment by whedon** [comment]
OK. v2.2.0 is the version.

**Comment by kthyng** [comment]
@pfeiffea You have a small typo in your title for the Zenodo archive (forbed instead of for bed) — can you fix this?

**Comment by pfeiffea** [comment]
Thanks for catching that @kthyng  

Fixed!

**Comment by kthyng** [comment]
@whedon set 10.5281/zenodo.4044683 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4044683" target="_blank">10.5281/zenodo.4044683</a> is the archive.

**Comment by kthyng** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1751

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1751, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1309193700883521540) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1752
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02341](https://doi.org/10.21105/joss.02341)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new publication @pfeiffea! Many thanks to reviewers @zsylvester and @ebgoldstein for your time, expertise, and hard work. We couldn't do this without you!

(I will leave this issue open until the DOI resolves)

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02341/status.svg)](https://doi.org/10.21105/joss.02341)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02341">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02341/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02341/status.svg
   :target: https://doi.org/10.21105/joss.02341
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02341/status.svg)](https://doi.org/10.21105/joss.02341)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
